ekm/signer_key_manager_test.go

```diff
@@ -788,3 +790,127 @@ func TestEkmListAccounts(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, 2, len(accounts))
 }
+
+func TestConcurrentSlashingProtectionAttData(t *testing.T) {
+	require.NoError(t, bls.Init(bls.BLS12_381))
+
+	km := testKeyManager(t, nil)
+
+	sk1 := &bls.SecretKey{}
+	require.NoError(t, sk1.SetHexString(sk1Str))
+	require.NoError(t, km.AddShare(sk1))
+
+	currentSlot := km.(*ethKeyManagerSigner).storage.Network().EstimatedCurrentSlot()
+	currentEpoch := km.(*ethKeyManagerSigner).storage.Network().EstimatedEpochAtSlot(currentSlot)
+
+	highestTarget := currentEpoch + minSPAttestationEpochGap + 1
+	highestSource := highestTarget - 1
+
+	attestationData := &phase0.AttestationData{
+		Slot:            currentSlot,
+		Index:           1,
+		BeaconBlockRoot: [32]byte{1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2},
+		Source: &phase0.Checkpoint{
+			Epoch: highestSource,
+			Root:  [32]byte{},
+		},
+		Target: &phase0.Checkpoint{
+			Epoch: highestTarget,
+			Root:  [32]byte{},
+		},
+	}
+
+	// Define function to concurrently attempt signing.
+	signAttestation := func(wg *sync.WaitGroup, errChan chan error) {
+		defer wg.Done()
+		_, _, err := km.(*ethKeyManagerSigner).SignBeaconObject(
+			attestationData,
+			phase0.Domain{},
+			sk1.GetPublicKey().Serialize(),
+			spectypes.DomainAttester,
+		)
+		errChan <- err
+	}
+
+	// Set up concurrency.
+	const goroutineCount = 100
+	var wg sync.WaitGroup
+	errChan := make(chan error, goroutineCount)
+
+	for i := 0; i < goroutineCount; i++ {
+		wg.Add(1)
+		go signAttestation(&wg, errChan)
+	}
+
+	// Wait for all goroutines to complete.
+	wg.Wait()
+	close(errChan)
+
+	// Count errors and successes.
+	var slashableErrors, successCount int
+	for err := range errChan {
+		if err != nil && err.Error() == "slashable attestation (HighestAttestationVote), not signing" {
+			slashableErrors++
+		} else if err == nil {
+			successCount++
+		}
```

Comment:
Shouldn't we check here for the absence of errors other than `"slashable attestation (HighestAttestationVote), not signing"` (or leave a comment explaining why not) ?  Same for another test.

Reply:
The purpose of this test is not to fail fast, but to ensure that under high contention, slashing protection works correctly without causing deadlocks or race conditions. We’re verifying that only one signing succeeds and the rest fail with the expected slashing error

Reply:
I mean, if we encounter any error other than `slashable attestation (HighestAttestationVote), not signing` here - there is no point running this test further (doesn't matter if it deadlocks or races, "functionality doesn't function" at all as we want it to).

Reply:
I see your point! I’ve adjusted the test to fail fast on unexpected error
https://github.com/ssvlabs/ssv/pull/1809/commits/d5c4d31d0d1d6799cd4db0a067cc9024567420e3